cloud/scope/powervs_machine.go

```diff
@@ -996,7 +998,45 @@ func (m *PowerVSMachineScope) CreateVPCLoadBalancerPoolMember(ctx context.Contex
 
 		internalIP := m.GetMachineInternalIP()
 
+		// lbAdditionalListeners is a mapping of additionalListener's port-protocol to the additionalListener as defined in the specification
+		// It will be used later to get the default pool associated with the listener
+		lbAdditionalListeners := map[string]infrav1beta2.AdditionalListenerSpec{}
+		for _, additionalListener := range lb.AdditionalListeners {
+			if additionalListener.Protocol == nil {
+				additionalListener.Protocol = &infrav1beta2.VPCLoadBalancerListenerProtocolTCP
+			}
+			lbAdditionalListeners[fmt.Sprintf("%d-%s", additionalListener.Port, *additionalListener.Protocol)] = additionalListener
+		}
+
+		// loadBalancerListeners is a mapping of the loadBalancer listener's defaultPoolName to the additionalListener
+		// as the default pool name might be empty in spec and should be fetched from the cloud's listener
+		loadBalancerListeners := map[string]infrav1beta2.AdditionalListenerSpec{}
+		for _, listener := range loadBalancer.Listeners {
+			listenerOptions := &vpcv1.GetLoadBalancerListenerOptions{}
+			listenerOptions.SetLoadBalancerID(*loadBalancer.ID)
+			listenerOptions.SetID(*listener.ID)
+			loadBalancerListener, _, err := m.IBMVPCClient.GetLoadBalancerListener(listenerOptions)
+			if err != nil {
+				return nil, fmt.Errorf("failed to list %s load balancer listener: %w", *listener.ID, err)
+			}
+			if additionalListener, ok := lbAdditionalListeners[fmt.Sprintf("%d-%s", *loadBalancerListener.Port, *loadBalancerListener.Protocol)]; ok {
+				if loadBalancerListener.DefaultPool != nil {
+					loadBalancerListeners[*loadBalancerListener.DefaultPool.Name] = additionalListener
+				}
+			} else if loadBalancerListener.Port != nil && *loadBalancerListener.Port == int64(6443) {
+				// add default pool 6443 to all control plane machines
+				log.V(3).Info("loadbalancerlisteners port match", "port", loadBalancerListener.Port)
+
+				protocol := infrav1beta2.VPCLoadBalancerListenerProtocol(*loadBalancerListener.Protocol)
+				listener := infrav1beta2.AdditionalListenerSpec{
+					Port:     *loadBalancerListener.Port,
+					Protocol: &protocol,
+				}
+				loadBalancerListeners[*loadBalancerListener.DefaultPool.Name] = listener
```

Comment:
Is it safe here to directly access `*loadBalancerListener.DefaultPool`, I see you have added a check in line no:1023